eldritch/items/unique.py

```diff
@@ -56,3 +58,228 @@ def get_trigger(self, event, owner, state):
+      return None
+
+    if self.exhausted:
+      return None
+
+    print(owner.movement_points)
+    if owner.movement_points < movement_cost:
+      return None
+
+    rolls = events.DiceRoll(owner, 2)
+    two_success = events.Draw(owner, "spells", 2, keep_count=2)
+    one_success = events.Sequence(
+        [events.Draw(owner, "spells", 1), events.Gain(owner, {"clues": 2})],
+        owner
+    )
+    no_success = events.Gain(owner, {"clues": 4})
+
+    return events.Sequence([
+        events.DiscardSpecific(owner, [self]),
+        events.ChangeMovementPoints(owner, -movement_cost),
+        rolls,
+        events.Conditional(
+            owner, rolls, "successes", {0: no_success, 1: one_success, 2: two_success}
+        )
+    ], owner)
+
+
+class BlueWatcher(Item):
+  def __init__(self, idx):
+    super().__init__("Blue Watcher", idx, "unique", {}, {}, None, 4)
+
+  def get_usable_interrupt(self, event, owner, state):
+    if not isinstance(event, (events.Check, events.CombatRound)):
+      return None
+    if not owner == event.character:
+      return None
+    if owner.stamina < 2:
+      return None
+    if isinstance(event, events.CombatRound):
+      return events.Sequence([
+          events.PassCombatRound(
+              event,
+              log_message="{char_name} passed a combat round against {monster_name} using Blue Watcher of the Pyramid"
+          ),
+          events.DiscardSpecific(owner, [self])
+      ], owner)
+    # TODO: Pass combat check or gate close check
+    if (event.check_type == "combat"
+        or (event.check_type in ("fight", "lore")
+            and isinstance(state.event_stack[-2], events.GateCloseAttempt))):
+      return events.Sequence(
+          [events.DiscardSpecific(owner, [self]), events.PassCheck(owner, event)],
+          owner
+      )
+    return None
+
+
+class EnchantedJewelry(Item):
+  def __init__(self, idx):
+    super().__init__("Enchanted Jewelry", idx, "unique", {}, {}, None, 3)
+    self.tokens["stamina"] = 0
+    self.max_tokens["stamina"] = 3
+
+  def get_usable_interrupt(self, event, owner, state):
+    if not isinstance(event, events.GainOrLoss):
+      return None
+
+    if not owner == event.character:
+      return None
+
+    loss = event.losses.get("stamina", 0)
+    if isinstance(loss, values.Value):
+      loss = loss.value(state)
+
+    if loss <= 0:
+      return None
+
+    reduction = events.LossPrevention(self, event, "stamina", 1)
+    return events.Sequence(
+        [
+            reduction,
+            events.AddToken(self, "stamina", event.character, n_tokens=1)
+        ],
+        owner
+    )
+
+
+class GateBox(Item):
+  def __init__(self, idx):
+    super().__init__("Gate Box", idx, "unique", {}, {}, None, 4)
+
+  def get_interrupt(self, event, owner, state):
+    if not isinstance(event, events.GateChoice) or not event.character == owner:
+      return None
+    if event.gate_name is None:
+      return None
+    return events.Sequence([
```

Comment:
I don't think this will do what you want. Events that reference the GateChoice don't have their references updated, so they will be referencing a cancelled GateChoice and do the wrong thing. Instead, you probably want to modify the GateChoice to set gate_name to None.